### PR TITLE
sequence: Initial sequence policy addition

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
Initial sequence policy addition

## Related Issues
n/a

## Checklist
- [na] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [na] Write tests
- [na] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

File: addOns/sequence/src/main/zapHomeFiles/policies/sequence.policy [hunk 1]
```diff
@@ -0,0 +1,313 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<configuration>
+    <policy>sequence</policy>
+    <scanner>
+        <level>MEDIUM</level>
```
Comment (thc202) [inline addOns/sequence/src/main/zapHomeFiles/policies/sequence.policy:5]: IMO this should be OFF and only meaningful rules be defined in the file.

File: addOns/sequence/CHANGELOG.md [hunk 2]
```diff
@@ -5,7 +5,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
 ### Added
-- Initial sequence-activeScan implementation.
+- Initial sequence-activeScan implementation
```
Comment (thc202) [inline addOns/sequence/CHANGELOG.md:8]: Dot removed.

File: addOns/sequence/CHANGELOG.md [hunk 3]
```diff
@@ -5,7 +5,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
...
+- Default Sequence active scan policy.
```
Comment (thc202) [inline addOns/sequence/CHANGELOG.md:9]: Technically it's not the default, it's not being used (by default) anywhere.

File: addOns/sequence/src/main/zapHomeFiles/policies/sequence.policy [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+<configuration>
+    <policy>sequence</policy>
+    <scanner>
+        <level>OFF</level>
+        <strength>MEDIUM</strength>
+    </scanner>
+    <plugins>
+        <p6>
+            <enabled>false</enabled>
+            <level>OFF</level>
```
Comment (psiinon) [inline addOns/sequence/src/main/zapHomeFiles/policies/sequence.policy:11]: Strictly speaking we dont need the `OFF` ones any more, as the default level is `OFF`.

File: addOns/sequence/src/main/zapHomeFiles/policies/sequence.policy [hunk 5]
```diff
@@ -0,0 +1,313 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<configuration>
+    <policy>sequence</policy>
+    <scanner>
+        <level>OFF</level>
+        <strength>MEDIUM</strength>
+    </scanner>
+    <plugins>
+        <p6>
```
Comment (psiinon) [inline addOns/sequence/src/main/zapHomeFiles/policies/sequence.policy:9]: Can we add a fake node or commend giving the scan rule name? Just for documentation purposes?
TBH I'd like to make that change to the core as well...

File: addOns/sequence/src/main/zapHomeFiles/policies/sequence.policy (hunk 4 above)
Comment (psiinon) [reply addOns/sequence/src/main/zapHomeFiles/policies/sequence.policy:11]: Can we get away wihout the `<enabled>false</enabled>` as well?

**Comment by kingthorin** [comment]
I'll see what I can do as far as adding a name element.

File: addOns/sequence/CHANGELOG.md (hunk 3 above)
Comment (thc202) [reply addOns/sequence/CHANGELOG.md]: Rather than change the changelog entry it should be changed the code to use it by default (IMO).

**Comment by psiinon** [comment]
The Sequence policy should be the recommended one, and I'm good with it being the default as well if one is not supplied

**Comment by kingthorin** [comment]
To be clear: Default to `Sequence.policy` if nothing defined/set, unless the user has configured `policyDefinition`?

```yaml
      policy:                                  # String: Name of the scan policy to be used, default: Default Policy.
    policyDefinition:                          # The policy definition - only used if the 'policy' is not set
```

So the text/help will need to be revised. Default Sequence.policy, used if no policyDefinition (my assumption being that if there is policyDefinition they went to that trouble and we therefore should not use the default policy)? Otherwise if there is a policy by some other name it will override policyDefinition?

File: addOns/sequence/CHANGELOG.md (hunk 3 above)
Comment (thc202) [reply addOns/sequence/CHANGELOG.md]: This will be handled as follow up.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**